samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -0,0 +1,743 @@
+
+        value_list = self.intrinsic_property_resolver(arguments[1], parent_function=IntrinsicResolver.FN_JOIN)
+
+        verify_intrinsic_type_list(value_list, IntrinsicResolver.FN_JOIN,
+                                   message="The list of values in {} after the "
+                                           "delimiter must be a list".format(IntrinsicResolver.FN_JOIN))
+
+        sanitized_value_list = [self.intrinsic_property_resolver(item, parent_function=IntrinsicResolver.FN_JOIN) for
+                                item in value_list]
+        verify_all_list_intrinsic_type(sanitized_value_list, verification_func=verify_intrinsic_type_str,
+                                       property_type=IntrinsicResolver.FN_JOIN)
+
+        return delimiter.join(sanitized_value_list)
+
+    def handle_fn_split(self, intrinsic_value):
+        """
+        { "Fn::Split" : [ "delimiter", "source string" ] }
+        This function will then split the source_string based on the delimiter
+
+        This intrinsic function will resolve all the objects within the function's value and check their type.
+        Parameter
+        ----------
+        intrinsic_value: list, dict
+            This is the value of the object inside the Fn::Split intrinsic function property
+
+        Return
+        -------
+        A string with the resolved attributes
+        """
+        arguments = self.intrinsic_property_resolver(intrinsic_value, parent_function=IntrinsicResolver.FN_SPLIT)
+
+        verify_intrinsic_type_list(arguments, IntrinsicResolver.FN_SPLIT)
+
+        delimiter = self.intrinsic_property_resolver(arguments[0], parent_function=IntrinsicResolver.FN_SPLIT)
+
+        verify_intrinsic_type_str(delimiter, IntrinsicResolver.FN_SPLIT, position_in_list="first")
+
+        source_string = self.intrinsic_property_resolver(arguments[1], parent_function=IntrinsicResolver.FN_SPLIT)
+
+        verify_intrinsic_type_str(source_string, IntrinsicResolver.FN_SPLIT, position_in_list="second")
+
+        return source_string.split(delimiter)
+
+    def handle_fn_base64(self, intrinsic_value):
+        """
+        { "Fn::Base64" : valueToEncode }
+        This intrinsic function will then base64 encode the string using python's base64.
+
+        This function will resolve all the intrinsic properties in valueToEncode
+        Parameter
+        ----------
+        intrinsic_value: list, dict
+           This is the value of the object inside the Fn::Base64 intrinsic function property
+
+        Return
+        -------
+        A string with the resolved attributes
+        """
+        data = self.intrinsic_property_resolver(intrinsic_value, parent_function=IntrinsicResolver.FN_BASE64)
+
+        verify_intrinsic_type_str(data, IntrinsicResolver.FN_BASE64)
+        # Encoding then decoding is required to return a string of the data
+        return base64.b64encode(data.encode()).decode()
```

Comment:
Are we sure this is the correct one? How did you validate this is correct

Reply:
I was following the spec from https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-base64.html. I also tested some base64 encoded responses.